hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/NetworkTransactionGetRecordHandler.java

```diff
@@ -58,13 +70,72 @@ public Response createEmptyResponse(@NonNull final ResponseHeader header) {
     @Override
     public void validate(@NonNull final QueryContext context) throws PreCheckException {
         requireNonNull(context);
-        throw new UnsupportedOperationException("Not implemented");
+        final var op = context.query().transactionGetRecordOrThrow();
+
+        // The transaction ID must be specified
+        if (!op.hasTransactionID()) throw new PreCheckException(INVALID_TRANSACTION_ID);
+
+        // The record must exist for that transaction ID
+        final var recordCache = context.createStore(RecordCache.class);
+        final var txnId = op.transactionIDOrThrow();
+        final var record = recordCache.getReceipt(txnId);
+        mustExist(record, INVALID_TRANSACTION_ID);
+
+        // verify that the account id exist and not default
```

Comment:
We could check if the account id is missing _before_ calling `recordCache.getReceipt(txnId)`, and that might save us a bit of work.

But a `TransactionID` with missing account id will never have a receipt in the cache.

So there's no use checking for `AccountID.DEFAULT` _after_`.

Reply:
fixed it you are right missed that.